### DISCUSSION TITLE
Orphaned Records

### DISCUSSION THREAD

**Discussion by adrianomic** [mastodon/mastodon#18779] [Q&A] [answered]
Let's say Status A has many favorites.

https://github.com/mastodon/mastodon/blob/fff0a55df215010abddb218b598049a6fb0ed669/app/services/batched_remove_status_service.rb#L30

If batched_remove_status_service is called, then there will be many orphaned records. Am I correct?

**Comment by nightpool** [comment, marked as answer]
No. Favourites column "status_id" has a foreign key constraint set to
"cascade", so removing the status row will automatically remove all
favorite rows. it's a little inefficient from a database transaction
perspective, since it requires a longer exclusive lock on the favorites
table, but it works
